src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -0,0 +1,506 @@
+
+        Returns:
+            DrivingField: the object representing the global drive for the
+                AnalogueHamiltonianSimulation object
+        """
+
+        time_points = self._get_sample_times(time_interval)
+
+        # scaling factor for amp and detuning converts MHz (PL input) to rad/s (upload units)
+        amplitude = self._convert_to_time_series(
+            pulse.amplitude, time_points, scaling_factor=2 * np.pi * 1e6
+        )
+        detuning = self._convert_to_time_series(
+            pulse.detuning, time_points, scaling_factor=2 * np.pi * 1e6
+        )
+        phase = self._convert_to_time_series(pulse.phase, time_points)
+
+        drive = DrivingField(amplitude=amplitude, detuning=detuning, phase=phase)
+
+        return drive
+
+    @staticmethod
+    def _result_to_sample_output(res):
+        """This function converts a single shot of the QuEra measurement results to
+        0 (ground), 1 (excited) and NaN (failed to measure) for all atoms in the result.
+
+        The QuEra results are summarized via 3 values: status, pre_sequence, and post_sequence.
+
+        Status is success or fail. The pre_sequence is 1 if an atom in the ground state was
+        successfully initialized, and 0 otherwise. The post_sequence is 1 if an atom in the
+        ground state was measured, and 0 otherwise. Comparison of pre_sequence and post_sequence
+        reveals one of 4 possible outcomes. The first two (initial measurement of 0) indicate a
+        failure to initialize correctly, and will yeild a NaN result. The second two are measurements
+        of the excited and ground state repsectively, and yield 1 and 0.
+
+        0 --> 0: NaN - Atom failed to be placed (no atom detected in the ground state either before or after)
+        0 --> 1: NaN - Atom failed to be placed (but was recaptured, or something else weird happened)
+        1 --> 0: 1 (Rydberg state) - atom measured in ground state before, but not after
+        1 --> 1: 0 (ground state) - atom measured in ground state both before and after
+        """
+
+        # if entire measurement failed, all NaN
+        if not res.status.value.lower() == "success":
+            return np.array([np.NaN for i in res.pre_sequence])
+
+        # if a single atom failed to initialize, NaN for that individual measurement
+        pre_sequence = [i if i else np.NaN for i in res.pre_sequence]
+
+        # set entry to 0 if ground state is measured, 1 if excited state is measured, NaN if measurement failed
+        return np.array(pre_sequence - res.post_sequence)
+
+
+class BraketAwsAhsDevice(BraketAhsDevice):
+    """Amazon Braket AHS device for hardware in PennyLane.
+
+    Args:
+        wires (int or Iterable[Number, str]]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers
+            (i.e., ``[-1, 0, 2]``) or strings (``['ancilla', 'q1', 'q2']``).
```

Comment:
I think we need to add `device_arn` here.

Reply:
Thanks, missed that!